frontend/api/auth/index.ts

```diff
@@ -0,0 +1,217 @@
+import passport from 'passport';
+import { sign } from 'jsonwebtoken';
+import express, { Express, Request, Response, NextFunction } from 'express';
+import { Strategy as FacebookStrategy } from 'passport-facebook';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+import { Strategy as TwitterStrategy } from 'passport-twitter';
+import type { Profile } from 'passport';
+import { prisma } from '../../src/server/prisma';
+import { assignAnswerToUser } from '../answers';
+import ms from 'ms';
+import { respond404 } from '../../src/server/errors';
+
+const app: Express = express();
+
+if (process.env.VERCEL_ENV === 'preview') {
+  process.env[
+    'BASE_URL'
+  ] = `https://volebni-kalkulacka-2022-git-${process.env.VERCEL_GIT_COMMIT_REF}-ceskodigital.vercel.app`;
+}
+
+const BASE_URL = process.env['BASE_URL'] || 'http://localhost:3000';
+
+const providers = {
+  facebook: {
+    strategy: () => {
+      return new FacebookStrategy(
+        {
+          clientID: process.env['FACEBOOK_CLIENT_ID'] as string,
+          clientSecret: process.env['FACEBOOK_CLIENT_SECRET'] as string,
+          profileFields: ['id', 'displayName', 'email', 'gender'],
+          callbackURL: `${BASE_URL}/api/auth/facebook/callback`,
```

Comment:
Hele možná jinak – pojďme pro ty callbacky udělat separátní ENV (`OAUTH_CALLBACK_URL`), která sice bude defaultovat na tu `PUBLIC_URL`, ale bude možné ji nastavit i jinak. Na Vercelu se dá potom jednoduše přehazovat branche nad jednou doménou (třeba `volebni-kalkulacka-staging.vercel.app`), takže by se to dalo potom testovat i nad různými branchemi a nebylo by třeba měnit ty OAuth configy.